src/engine/Interfaces/Defaultable.ts

```diff
@@ -0,0 +1,21 @@
+import { extend } from '.././Util/Util';
+
+export abstract class Defaultable<T> {
```

Comment:
I think it may be possible to replace `T` with `this` type to refer back to the class itself, reducing need for generic arguments. Not sure though.